src/import/ht_hypertable_modify.c

```diff
@@ -459,6 +459,12 @@ lmerge_matched:;
 	 * visible to our MVCC snapshot.
 	 */
 
+#if PG17_GE
+	if (oldtuple != NULL)
+		ExecForceStoreHeapTuple(oldtuple, resultRelInfo->ri_oldTupleSlot,
+								false);
+	else
+#endif
```

Comment:
Looks like this code in upstream was added by this commit https://github.com/postgres/postgres/commit/5f2e179bd31e5f5803005101eb12a8d7bf8db8f3 so I think there are more things to import, no?

Also what this new code is fixing cause seems our regression tests are not reaching it per codecov?

Reply:
I was having a close look on it and actually we need to introduce support for `MERGE ... WHEN NOT MATCHED BY SOURCE THEN ...` for hypertables instead and it will require more work on ht_hypertable_modify.c

Reply:
I think this is mostly to silence coverity

Reply:
Looks like the coverity was already silenced, no? I mean it will no hurt to merge this tiny PR but it still requires more work to sync it with upstream. I'm working on another PR to make this new merge clause work for hypertables because I want to use it on the new cagg refresh code using merge statement.

Reply:
A [previous commit](https://github.com/timescale/timescaledb/commit/d7513b26f667fcdb269abf1ba0453ab28e0e3801)  I made added support for `MERGE ... RETURNING ...`. 
While working on the coverity fix, I realized that this code handling `oldtuple` was not added; that's what this fixes.

Reply:
@kpan2034 would be good then mention it on the commit message and in the PR description to make clear the change you're proposing.